test/surface/formatter_test.exs

```diff
@@ -1606,4 +1606,31 @@ defmodule Surface.FormatterTest do
     } />
     """)
   end
+
+  test "newlines without trailing whitespace in formatted attribute expressions" do
+    # This demonstrates a bugfix where the empty newline in between case clauses
+    # would be indented with spaces even though there were no contents on the line.
+    assert_formatter_outputs(
+      ~S"""
+      <Component bool_prop attribute={case @foo do
+        "bar" ->
+          true
+        "baz" ->
+          false
+      end} />
+      """,
+      ~S"""
+      <Component
+        bool_prop
+        attribute={case @foo do
+          "bar" ->
+            true
+
```

Comment:
Prior to this bugfix, this blank line would include a number of spaces depending on the indent level.